evalml/model_understanding/_partial_dependence_utils.py

```diff
@@ -307,6 +374,16 @@ def _partial_dependence(
             range of values to use in partial dependence. If custom_range is specified,
             the percentile + interpolation procedure is skipped and the values in custom_range
             are used.
+        fast_mode (bool, optional): Whether or not performance optimizations should be
+            used. Defaults to False. When True, copies of pipelines will be used to transform just the
+            column(s) we're calculating partial dependence for. This means that any pipeline
+            containing a component that relies on multiple columns for fit and transform should
+            not be used. See the ``_can_be_used_for_fast_partial_dependence`` property on components
+            to determine which components cannot be used for fast mode.
+        X_train (pd.DataFrame, np.ndarray): The data that was used to train the original pipeline. Will
+            be used in fast mode to train the cloned pipelines.
+        y_train (pd.Series, np.ndarray): The target data that was used to train the original pipeline. Will
+            be used in fast mode to train the cloned pipelines.
```

Comment:
Nit: another case where the docstring order doesn't match the argument order! It'd also be nice to add that X_train and y_train default to None here.